ui/src/pages/AllStories/index.tsx

```diff
@@ -159,7 +163,7 @@ export const AllStories: React.FC = () => {
   };
 
   const handleSwitchChange = (e, story) => {
-    dispatch({ type: "HANDLE_SWITCH_CHANGE", e, story });
+    dispatch({ type: "HANDLE_SWITCH_CHANGE", e, story: { ...story, is_visible: !story.is_visible }});
```

Comment:
With this change, changedVisibility would have the accurate is_visible value. Can we remove visibleState then and have `story.is_visible` instead of `state.visibleState.includes(d.ID)` in pendingMapChanges specifically on line 538 and 597. Apparently, I could not add a comment there since i think you can only add comments on changed parts of the file.